packages/arb-token-bridge-ui/src/components/common/WalletConnectChainDropdown.tsx

```diff
@@ -0,0 +1,66 @@
+import { Chain } from 'wagmi'
+import { useMemo } from 'react'
+
+import { TargetChainKey } from '../../util/WalletConnectUtils'
+import { getSupportedNetworks } from '../../util/networks'
+import { getWagmiChain } from '../../util/wagmi/getWagmiChain'
+import { NetworkListbox } from '../TransferPanel/NetworkListbox'
+
+export default function WalletConnectChainDropdown() {
+  const options = getSupportedNetworks(undefined, true)
+    .map(chainId => {
+      const wagmiChain = getWagmiChain(chainId)
+      const targetChainKey =
+        TargetChainKey[wagmiChain.name as keyof typeof TargetChainKey]
+
+      if (!targetChainKey) {
+        return undefined
+      }
+      return { ...wagmiChain, network: targetChainKey as string }
+    })
+    .filter((chain): chain is Chain => chain !== undefined)
+
+  const searchParams = useMemo(() => {
+    if (typeof window === 'undefined') return undefined
+    return new URLSearchParams(window.location.search)
+  }, [])
+
+  const walletConnectChain = searchParams?.get('walletConnectChain')
+
+  const chain = (() => {
```

Comment:
why do we need try catch?

Reply:
oops i was testing something else and didn't remove it